### ISSUE TITLE
Should we have Continuous Delivery from master branch to a NuGet feed?

### ISSUE THREAD

**Issue by eloekset** [fluentmigrator/fluentmigrator#1154]
As discussed in the [last part of the conversation](https://github.com/fluentmigrator/fluentmigrator/issues/1146#issuecomment-576365669) of issue #1146 continuous delivery to a NuGet feed from the master branch currently fails, because of the versioning scheme we get from the GitVersion tool doesn't make unique versions on each build.

We need to decide if we should have CD from master or not, and if we are, how should we change the versioning scheme to something that fits NuGet. Our old Travis pipeline doesn't push NuGet packages from the master branch, so it has been introduced with the switch to Azure Pipelines, simply because nothing has been done to avoid it (yet).

This is the old Travis build definition, which is still alive:
![image](https://user-images.githubusercontent.com/791721/72756116-9527c480-3bcc-11ea-8cb7-fa9c33d0898c.png)

**Comment by eloekset** [comment]
If we're going to follow [this GitFlow strategy](https://datasift.github.io/gitflow/IntroducingGitFlow.html), we're supposed to create **release/*** branches from develop and merge to **master** and then create a tag, i.e. **v4.0.0**, **v4.1.0** etc to do official releases. 

Then, I suppose, GitVersion will automatically make pre-release version numbers for our NuGet packages from the **release/*** branches and official version numbers from the tags that we call **vM.m.b**. 

That's maybe the purpose of the Travis configuration shown above. It wouldn't push any NuGet packages built on the **master** branch, unless there's a tag matching that version number pattern. But we could have been used to push manually packages built from **master** for all I know.

**Comment by eloekset** [comment]
I've now tested this in a separate [GitHub repo](https://github.com/eloekset/AzurePipelinesTest/releases) and got it working exactly like described in the previous comment.

So now I've updated both build and release pipelines for FluentMigrator to work like this as well.

We build for the master branch, so we'll get NuGet packages in the build artifacts, but we don't trigger the release pipeline, so those NuGet packages won't be pushed anywhere.
![image](https://user-images.githubusercontent.com/791721/74775216-8bf64a00-5295-11ea-99f4-bd43853fedd7.png)

The NuGet packages that will be pushed to NuGet feeds are those created from tags on the version format **vM.m.*** including pre-release version like those built from **develop** and **release/*** branches.
![image](https://user-images.githubusercontent.com/791721/74775418-f6a78580-5295-11ea-85a0-9041576d45e6.png)

So the next time a GitHub release is created with a version tag e.g. **v3.2.3**, it will be automatically pushed to our Azure Artifacts feed, and it should never break because of multiple packages with the same version.

Then it's just left to actually push to **nuget.org** from the release pipeline. That release stage is still just running some dummy echo scripts.
